File: C#JAVA_Client_Server/ClientServcerC#Protocols/Solution1/Domain/Validators/MasinaValidator.cs
using System;
using Domain.validators.interfaceAndException;

namespace Domain.validators
{
    public class MasinaValidator : IValidator<Masina>
    {
        public void validate(Masina entity)
        {
            if (entity.NrMasina.Length == 0 || entity.CapacitateMotor.Equals(null))
            {
                throw new Exception("Masina INVALIDA!");
            }
        }
    }
}

File: C#JAVA_Client_Server/ClientServcerC#Protocols/LongExample/Domain/src/main/java/entities/domain/validator/MasinaValidator.java
package entities.domain.validator;


import entities.domain.Masina;
import entities.domain.validator.interfaceAndExceptions.ValidationException;
import entities.domain.validator.interfaceAndExceptions.Validator;

public class MasinaValidator implements Validator<Masina> {
    @Override
    public void validate(Masina entity) throws ValidationException {
        if(entity.getNrMasina().length()==0 || entity.getCapacitateMotor()==null){
            throw new ValidationException("entities.domain.Masina INVALIDA");
        }
    }
}
